app/src/main/java/com/hadiyarajesh/notex/ui/testreminder/ReminderActivity.kt

```diff
@@ -0,0 +1,105 @@
+package com.hadiyarajesh.notex.ui.testreminder
+
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.work.*
+import com.hadiyarajesh.notex.databinding.ActivityReminderBinding
+import com.hadiyarajesh.notex.workmanager.ReminderWorker
+import java.text.SimpleDateFormat
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+class ReminderActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityReminderBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityReminderBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+
+        binding.btnpickup.setOnClickListener {
+            showDateTimePicker()
+        }
+        binding.btnsetreminder.setOnClickListener {
+
+            val todayDateTime = Calendar.getInstance()
+
+            var delayInSeconds = (date.timeInMillis) - todayDateTime.timeInMillis
+
+            Log.i("MYTAG", "delay=$delayInSeconds")
+
+            val formatter = SimpleDateFormat("dd.MM.yyyy hh:mm aa")
+
+            val fdate: String = formatter.format(date.timeInMillis)
+            binding.scheduledtime.setText(fdate)
+            createWorkRequest(
+                "Ui Design",
+                "Its time to build notifications design and delete notes design in Jetpack compose",
+                delayInSeconds,
+                3
+            )
+        }
+    }
+
+
+    private fun createWorkRequest(
+        title: String,
+        message: String,
+        timeDelayInSeconds: Long,
+        notiid: Int
+    ) {
+        val myWorkRequest: OneTimeWorkRequest = OneTimeWorkRequestBuilder<ReminderWorker>()
+            .setInitialDelay(timeDelayInSeconds, TimeUnit.MILLISECONDS)
+            .setInputData(
+                workDataOf(
+                    "title" to title,
+                    "message" to message,
+                    "notiid" to notiid
+
+
+                )
+            )
+            .build()
+
+        WorkManager.getInstance(this).enqueueUniqueWork(
+            "Reminder Worker", ExistingWorkPolicy.REPLACE,
+            myWorkRequest
+        )
+    }
+
+
+    lateinit var date: Calendar
+
+
+    fun showDateTimePicker() {
+        val currentDate: Calendar = Calendar.getInstance()
+        date = Calendar.getInstance()
+        DatePickerDialog(
```

Comment:
Make this a Generic function to show the DatePickerDialog based on the requirement. 

Always try to write the code in some generic way so that in future a fundamental change in the params can satisfy your requirement.

Reply:
@sumanabhi correct
@itxaliraza avoid this `Calendar` things and try to use java.time api (available since java 8). 
Something like this
```kotlin
 val date =  = LocalDateTime.now();  
```